soc/riscv/riscv-privileged/sifive-freedom/Kconfig.series

```diff
@@ -7,5 +7,6 @@ config SOC_SERIES_RISCV_SIFIVE_FREEDOM
 	bool "SiFive Freedom SOC implementation"
 	select RISCV
 	select SOC_FAMILY_RISCV_PRIVILEGED
+	select ARCH_HAS_RAMFUNC_SUPPORT
```

Comment:
Shouldn't this be defined in `arch/riscv/Kconfig.core`?

Reply:
That was the first place I looked, but there are more soc's defined in `soc/riscv`.
In my case I looked for a board with qemu support with `CONFIG_XIP=y`, then it would be `soc/riscv/riscv-privileged/Kconfig` with `CONFIG_SOC_FAMILY_RISCV_PRIVILEGED`.

But you are right, it does make sense to define it at a higher level with `select ARCH_HAS_RAMFUNC_SUPPORT if XIP`.
Should I add it on every derivative of RISC-V or is there a better place to define it global for all RISC-V soc's?

Reply:
I added it to `CONFIG_RISCV`.

Reply:
I enabled it now only for `CONFIG_SOC_FAMILY_RISCV_PRIVILEGED`.
To enable it for all RISC-V would need some adaption in the linker script for `esp32c3` (`soc/riscv/espressif_esp32/esp32c3/default.ld`).